src/core/protocol-authorization.ts

```diff
@@ -579,6 +582,31 @@ export class ProtocolAuthorization {
     throw new DwnError(DwnErrorCode.ProtocolAuthorizationActionNotAllowed, `inbound message action not allowed for author`);
   }
 
+  /**
+   * Verifies that writes adhere to the $size constraints if provided
+   * @throws {Error} if size is exceeded.
+   */
+  private static verifySizeLimit(
+    incomingMessage: RecordsWrite,
+    inboundMessageRuleSet: ProtocolRuleSet
+  ): void {
+    const { min = 0, max } = inboundMessageRuleSet.$size || {};
+
+    const messageSize = incomingMessage.message.descriptor.dataSize;
+
+    if (messageSize < min) {
+      throw new DwnError(DwnErrorCode.ProtocolAuthorizationSizeInvalid, `message size ${messageSize} is less than allowed ${min}`);
```

Comment:
Change to "Data size" instead of message size. We conventionally refer "Message" as the JSON object.

Reply:
👍